### PR TITLE
Sample bot for Inline queries

### PR DESCRIPTION
**Pull request by celestix**
# What

Describe the intent of your changes in detail, and what they are resolving. 

Provide links to external sources if relevant (eg: github issues, external discussions, etc)

# Impact

- Are your changes backwards compatible?
- Have you included documentation, or updated existing documentation?
- Do errors and log messages provide enough context?

### CODE REVIEW

**Comment by celestix** [comment]
@PaulSonOfLars sir

**Comment by celestix** [comment]
:/

File: samples/inlineQueryBot/main.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	if err != nil {
+		panic("failed to create new bot: " + err.Error())
+	}
+
+	// Create updater and dispatcher.
+	updater := ext.NewUpdater(nil)
+	dispatcher := updater.Dispatcher
+
+	dispatcher.AddHandler(handlers.NewCommand("start", start))
+	dispatcher.AddHandler(handlers.NewInlineQuery(inlinequery.Query("gotgbot"), startIQ))
```
Comment (PaulSonOfLars) [inline samples/inlineQueryBot/main.go:30]: Its not clear to someone with no inline query experience what this does. How is it triggered? For the sake of this example, does it not make more sense to use `inlinequery.All()`?

Try adding comments, maybe different IQ examples, explanations of how each one works.

File: samples/inlineQueryBot/main.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	"os"
+
+	"github.com/PaulSonOfLars/gotgbot/v2"
+	"github.com/PaulSonOfLars/gotgbot/v2/ext"
+	"github.com/PaulSonOfLars/gotgbot/v2/ext/handlers"
+	"github.com/PaulSonOfLars/gotgbot/v2/ext/handlers/filters/inlinequery"
+)
+
+func main() {
+	// Create bot from environment value.
```
Comment (PaulSonOfLars) [inline samples/inlineQueryBot/main.go:15]: This doesnt just need any bot; it needs to be a bot with inline queries enabled in botfather. Would recommend adding that for clarity.

File: samples/inlineQueryBot/main.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	if err != nil {
+		panic("failed to start polling: " + err.Error())
+	}
+	fmt.Printf("%s has been started...\n", b.User.Username)
+
+	// Idle, to keep updates coming in, and avoid bot stopping.
+	updater.Idle()
+}
+
+// start introduces the bot
```
Comment (PaulSonOfLars) [inline samples/inlineQueryBot/main.go:43]: This comment is not accurate; this doesnt just introduce the bot, it also links to two of the things you created this PR for; to demonstrate inline capabilities.
The comment should be expanded to explain the functionality of each button you've added.

File: samples/inlineQueryBot/main.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+			}},
+		},
+	})
+	if err != nil {
+		fmt.Println("failed to send: " + err.Error())
+	}
+	return nil
+}
+
+func startIQ(b *gotgbot.Bot, ctx *ext.Context) error {
```
Comment (PaulSonOfLars) [inline samples/inlineQueryBot/main.go:62]: Comment to explain what this function does? 
And what happens when the result is clicked?

File: samples/inlineQueryBot/main.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	iq := ctx.InlineQuery
```
Comment (PaulSonOfLars) [inline samples/inlineQueryBot/main.go:63]: is this variable necessary here? its only used twice

File: samples/inlineQueryBot/main.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+	// Idle, to keep updates coming in, and avoid bot stopping.
+	updater.Idle()
+}
+
+// start introduces the bot
+func start(b *gotgbot.Bot, ctx *ext.Context) error {
+	query := "gotgbot"
+	_, err := ctx.EffectiveMessage.Reply(b, fmt.Sprintf("Hello, I'm @%s. I'm a sample bot for inline queries.", b.User.Username), &gotgbot.SendMessageOpts{
+		ParseMode: "html",
```
Comment (PaulSonOfLars) [inline samples/inlineQueryBot/main.go:47]: why HTML? you're not using any formatting

File: samples/inlineQueryBot/main.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+				MessageText: "Press the following link to open gotgbot's github link.",
+			},
+			ReplyMarkup: &gotgbot.InlineKeyboardMarkup{
+				InlineKeyboard: [][]gotgbot.InlineKeyboardButton{{
+					{Text: "Press me", Url: "github.com/paulSonOfLars/gotgbot"},
+				}},
+			},
+		},
+	}
+	iq.Answer(b, result, nil)
```
Comment (PaulSonOfLars) [inline samples/inlineQueryBot/main.go:79]: Is it worth handling this error?

**Review by PaulSonOfLars — COMMENTED** [review]
Hi, thank you for the PR, and sorry for the late reply!
In the future, please keep in mind that meaningful reviews take time, and time might be difficult to find. No need to repeatedly comment, especially for non-urgent PRs; I'll get to it as soon as possible.

PR looks good overall, but I would strongly recommend adding more comments and explanations as to what each thing does, and what the requirements are. Having never written any inlinequery code before, I found some aspects confusing and wasnt sure how things worked together. I imagine this might be the case for many others too.

**Pull request closed without merging**